pkg/zfs/xfs_util.go

```diff
@@ -29,35 +30,54 @@ func xfsTempMount(volume string) error {
 	device := ZFSDevPath + volume
 	pvol := strings.Split(volume, "/")
 
+	// evaluate the symlink to get the dev path for volume
+	dev, err := filepath.EvalSymlinks(device)
+	if err != nil {
+		return err
+	}
+
 	// create a temporary directory to mount the xfs file system
 	tmpdir := "/tmp/" + pvol[1]
-	err := os.Mkdir(tmpdir, 0755)
-	if err != nil {
+	err = os.Mkdir(tmpdir, 0755)
+	if os.IsNotExist(err) {
 		klog.Errorf("xfs: failed to create tmpdir %s error: %s", tmpdir, err.Error())
 		return err
 	}
 
-	// mount with nouuid, so that it can play the log
-	cmd := exec.Command("mount", "-o", "nouuid", device, tmpdir)
-	out, err := cmd.CombinedOutput()
+	/*
+	 * Device might have already mounted at the tmp path but umount might have failed
+	 * in previous attempt. Checking here if device is not mounted then only attempt
+	 * to mount it, otherwise proceed with the umount.
+	 */
+	curMounts, err := GetMounts(dev)
 	if err != nil {
-		klog.Errorf("xfs: failed to mount volume %s=>%s error: %s", device, tmpdir, string(out))
+		klog.Errorf("xfs: get mounts failed dev: %s err: %v", device, err.Error())
```

Comment:
is it possible to check if the mount has the nouuid option?

Reply:
Why do we need that? Can you elaborate a little bit? The driver makes sure that the volume can not be mounted by the pod until this operation completes successfully so it is not possible that the volume is mounted by someone in the meanwhile.
Also we are using TMP path for mounting it, which will make sure that it is mounted by this guy only otherwise mount/umount will fail for TMP path.

Reply:
First off, I missed that this is a temporary mount. Was thinking of a stale mount case from the previous try - that was probably mounted without adding nouuid option. 

Is this the flow:
- clone with pv-uuidx
- mount /tmp/pv-uuidx (with nouuid)
- unmount /tmp/pv-uuidx
- proceed with mounting to pod/pv-uuidx

Reply:
Yes @kmova , that is correct. while creating clone only we generate the new UUID. To do that  mount it at the temp location to replay the log.